diff --git a/old/interpreters/quickjs/CMakeLists.txt b/new/interpreters/quickjs/CMakeLists.txt
--- a/old/interpreters/quickjs/CMakeLists.txt
+++ b/new/interpreters/quickjs/CMakeLists.txt
@@ -29,11 +29,15 @@ if(CONFIG_INTERPRETERS_QUICKJS)
     FetchContent_Declare(
       quickjs_fetch
       URL ${QUICKJS_URL_BASE}/6e2e68fd0896957f92eb6c242a2e048c1ef3cae0.zip
+          URL_HASH
+          SHA256=d4542882686eefa8c0c80493dfb30d858588c0553ffe17a058189132607ff24e
           SOURCE_DIR ${CMAKE_CURRENT_LIST_DIR}/quickjs BINARY_DIR
           ${NUTTX_BINARY_DIR}/apps/interpreters/quickjs/quickjs
       PATCH_COMMAND
-        patch -p1 -d ${CMAKE_CURRENT_LIST_DIR}/quickjs <
+        patch -p1 -d ${CMAKE_CURRENT_LIST_DIR}/quickjs -i
         ${CMAKE_CURRENT_LIST_DIR}/0001-Disabled-unsupported-feature-on-NuttX.patch
+        COMMAND patch -p1 -d ${CMAKE_CURRENT_LIST_DIR}/quickjs -i
+                ${CMAKE_CURRENT_LIST_DIR}/0003-Avoid-NuttX-list-macro-collisions.patch
       DOWNLOAD_NO_PROGRESS true
       TIMEOUT 30)
 
@@ -73,6 +77,9 @@ if(CONFIG_INTERPRETERS_QUICKJS)
     -Wno-array-bounds
     -Wno-incompatible-pointer-types
     -Wno-implicit-function-declaration
+    -Wno-strict-prototypes
+    -Wno-unused-but-set-variable
+    -Wno-unused-variable
     -Wno-unused-function
     -Wno-format
     -D__linux__
@@ -92,7 +99,8 @@ if(CONFIG_INTERPRETERS_QUICKJS)
     ${QUICKJS_DIR}/libregexp.c
     ${QUICKJS_DIR}/libbf.c
     ${QUICKJS_DIR}/libunicode.c
-    ${QUICKJS_DIR}/cutils.c)
+    ${QUICKJS_DIR}/cutils.c
+    ${CMAKE_CURRENT_LIST_DIR}/qjs_nuttx_compat.c)
 
   if(CONFIG_ARCH_ARM)
     list(APPEND QUICKJS_FLAGS -DFE_TONEAREST=0x00000000)
@@ -101,6 +109,13 @@ if(CONFIG_INTERPRETERS_QUICKJS)
     list(APPEND QUICKJS_FLAGS -DFE_TOWARDZERO=0x00c00000)
   endif()
 
+  if(CONFIG_ARCH_RISCV)
+    list(APPEND QUICKJS_FLAGS -DFE_TONEAREST=0x00000000)
+    list(APPEND QUICKJS_FLAGS -DFE_UPWARD=0x00000003)
+    list(APPEND QUICKJS_FLAGS -DFE_DOWNWARD=0x00000002)
+    list(APPEND QUICKJS_FLAGS -DFE_TOWARDZERO=0x00000001)
+  endif()
+
   if(CONFIG_INTERPRETERS_QUICKJS_BIGNUM)
     list(APPEND QUICKJS_FLAGS -DCONFIG_BIGNUM)
     list(APPEND QUICKJS_CSRCS ${CMAKE_CURRENT_BINARY_DIR}/qjscalc.c)
