diff --git a/old/arch/risc-v/src/esp32p4/hal_esp32p4.cmake b/new/arch/risc-v/src/esp32p4/hal_esp32p4.cmake
--- a/old/arch/risc-v/src/esp32p4/hal_esp32p4.cmake
+++ b/new/arch/risc-v/src/esp32p4/hal_esp32p4.cmake
@@ -562,6 +562,14 @@ if(CONFIG_ESPRESSIF_MIPI_CSI)
     ${ESP_HAL_3RDPARTY_REPO}/components/esp_hw_support/mipi_csi_share_hw_ctrl.c
     ${ESP_HAL_3RDPARTY_REPO}/components/hal/color_hal.c
     ${ESP_HAL_3RDPARTY_REPO}/components/esp_hal_cam/isp_hal.c)
+
+  target_include_directories(
+    arch
+    PRIVATE ${ESP_HAL_3RDPARTY_REPO}/components/upper_hal_cam/include
+            ${ESP_HAL_3RDPARTY_REPO}/components/upper_hal_cam/interface
+            ${ESP_HAL_3RDPARTY_REPO}/components/upper_hal_cam/csi/include
+            ${ESP_HAL_3RDPARTY_REPO}/components/upper_hal_cam/csi/src
+            ${ESP_HAL_3RDPARTY_REPO}/components/upper_hal_isp/include)
 endif()
 
 if(CONFIG_ESPRESSIF_IDF_ENV_FPGA)
