diff --git a/old/system/desktop/CMakeLists.txt b/new/system/desktop/CMakeLists.txt
--- a/old/system/desktop/CMakeLists.txt
+++ b/new/system/desktop/CMakeLists.txt
@@ -40,7 +40,14 @@ if(CONFIG_SYSTEM_DESKTOP)
     homeassistant_resource.c
     INCLUDE_DIRECTORIES
     ${NUTTX_APPS_DIR}/interpreters/quickjs/quickjs
+    COMPILE_FLAGS
+    -Wa,-I,${CMAKE_CURRENT_LIST_DIR}
     DEPENDS
     libqjs
     lvgl)
+
+  set_source_files_properties(
+    desktop_font.c
+    PROPERTIES OBJECT_DEPENDS
+               "${CMAKE_CURRENT_LIST_DIR}/AlibabaPuHuiTi-3-55-Regular.otf")
 endif()
