diff --git a/nuttx/src/platform/os.c b/nuttx/src/platform/os.c
index 68acacf53..b5bec24ec 100644
--- a/nuttx/src/platform/os.c
+++ b/nuttx/src/platform/os.c
@@ -27,6 +27,8 @@
 #include <nuttx/kmalloc.h>
 #include <semaphore.h>
 #include <unistd.h>
+#include <fcntl.h>
+#include <nuttx/kthread.h>
 #include <sys/types.h>
 #include <nuttx/sched.h>
 #include <nuttx/mm/mm.h>
@@ -156,6 +158,12 @@ typedef struct intr_handle_data_t intr_handle_data_t;
 
 typedef struct non_shared_isr_arg_t non_shared_isr_arg_t;
 
+struct intr_adapter_to_nuttx
+{
+  esp_os_intr_handler_t handler;
+  FAR void *arg;
+};
+
 struct non_shared_isr_arg_t
 {
   intr_handler_t isr;
@@ -1342,36 +1350,14 @@ int esp_os_create_task_pinned_to_core(esp_os_task_function_t task_func,
   argv[0] = ptr_buf;
   argv[1] = NULL;
 
-  tcb = kmm_zalloc(sizeof(struct tcb_s));
-  if (tcb == NULL)
-    {
-      kmm_free(wrapper_args);
-      return -1;
-    }
-
-  tcb->flags = TCB_FLAG_TTYPE_KERNEL | TCB_FLAG_FREE_TCB;
-
-  ret = nxtask_init(tcb, name, priority,
-                    NULL, stack_size,
-                    task_wrapper_entry, argv, NULL, NULL);
+  ret = kthread_create(name, priority, stack_size, task_wrapper_entry, argv);
   if (ret < 0)
     {
       kmm_free(wrapper_args);
-      kmm_free(tcb);
       return -1;
     }
 
-#ifdef CONFIG_SMP
-  if (core_id >= 0)
-    {
-      cpu_set_t cpuset;
-      CPU_ZERO(&cpuset);
-      CPU_SET(core_id, &cpuset);
-      tcb->affinity = cpuset;
-    }
-#endif
-
-  if (task_notify_register(tcb->pid) == NULL)
+  if (task_notify_register(ret) == NULL)
     {
       _warn("esp_os_create_task: out of memory for notify entry '%s'\n",
             name);
@@ -1379,10 +1365,9 @@ int esp_os_create_task_pinned_to_core(esp_os_task_function_t task_func,
 
   if (task_handle != NULL)
     {
-      *task_handle = (esp_os_task_handle_t)tcb->pid;
+      *task_handle = (esp_os_task_handle_t)ret;
     }
 
-  nxtask_activate(tcb);
   return 0;
 }
 
