diff --git a/boards/risc-v/esp32p4/common/scripts/esp32p4_sections.rev3.ld b/boards/risc-v/esp32p4/common/scripts/esp32p4_sections.rev3.ld
--- a/boards/risc-v/esp32p4/common/scripts/esp32p4_sections.rev3.ld
+++ b/boards/risc-v/esp32p4/common/scripts/esp32p4_sections.rev3.ld
@@ -58,9 +58,11 @@
 #if CONFIG_P4_REV3_MSPI_CRASH_AFTER_POWER_UP_WORKAROUND
   .rtc.p4_rev3_mspi_workaround :
   {
-    ALIGNED_SYMBOL(4, _rtc_p4_rev3_mspi_workaround_start)
+    . = ALIGN(4);
+    _rtc_p4_rev3_mspi_workaround_start = ABSOLUTE(.);
     KEEP (*(.p4_rev3_mspi_workaround.rtc_text .p4_rev3_mspi_workaround.rtc_text.*))
-    ALIGNED_SYMBOL(4, _rtc_p4_rev3_mspi_workaround_end)
+    . = ALIGN(4);
+    _rtc_p4_rev3_mspi_workaround_end = ABSOLUTE(.);
   } > rev3_mspi_workaround_seg
 #endif
 
diff --git a/arch/risc-v/src/esp32p4/hal_esp32p4.mk b/arch/risc-v/src/esp32p4/hal_esp32p4.mk
--- a/arch/risc-v/src/esp32p4/hal_esp32p4.mk
+++ b/arch/risc-v/src/esp32p4/hal_esp32p4.mk
@@ -299,2 +299,7 @@
 CHIP_ASRCS += chip$(DELIM)$(ESP_HAL_3RDPARTY_REPO)$(DELIM)components$(DELIM)esp_hw_support$(DELIM)lowpower$(DELIM)port$(DELIM)$(CHIP_SERIES)$(DELIM)sleep_cpu_asm.S
+ifeq ($(CONFIG_P4_REV3_MSPI_CRASH_AFTER_POWER_UP_WORKAROUND),y)
+CHIP_ASRCS += chip$(DELIM)$(ESP_HAL_3RDPARTY_REPO)$(DELIM)components$(DELIM)esp_hw_support$(DELIM)lowpower$(DELIM)port$(DELIM)$(CHIP_SERIES)$(DELIM)p4_rev3_mspi_workaround.S
+LDFLAGS += -u p4_rev3_mspi_workaround
+endif
+
 CHIP_CSRCS += chip$(DELIM)$(ESP_HAL_3RDPARTY_REPO)$(DELIM)components$(DELIM)esp_hw_support$(DELIM)lowpower$(DELIM)port$(DELIM)$(CHIP_SERIES)$(DELIM)sleep_cpu.c
